core/src/main/java/org/jruby/util/io/ChannelFD.java

```diff
@@ -131,7 +130,7 @@ private void initChannelTypes() {
         else chRead = null;
         if (ch instanceof WritableByteChannel) chWrite = (WritableByteChannel)ch;
         else chWrite = null;
-        if (ch instanceof SeekableByteChannel) chSeek = (SeekableByteChannel)ch;
```

Comment:
OK, reinstated the use of SeekableByteChannel and puts a copy of the ChannelFD source in Ruboto.